### PR TITLE
Add bpe tokenization

### PR DESCRIPTION
**Pull request by SpirinEgor**

### CODE REVIEW

File: src/data/preprocess/bpe_tokenizer.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    bpe_tokenizer.normalizer = Sequence([NFKC(), Lowercase(), StripAccents()])
+    bpe_tokenizer.pre_tokenizer = ByteLevel()
+
+    trainer = BpeTrainer(vocab_size=VOCAB_SIZE, special_tokens=SPECIAL_TOKENS)
+
+    with open(vocabulary_path, "rb") as vf:
+        token_counter = pickle.load(vf)
+
+    subtokens = list(token_counter.keys())
+    bpe_tokenizer.train_from_iterator(batch_iterator(subtokens), trainer=trainer, length=len(subtokens))
```
Comment (SokolovYaroslav) [inline src/data/preprocess/bpe_tokenizer.py]: Хм, это ведь не учитывает частотность слов. Получается, что BPE не так хорошо сжимает, как мог бы

File: src/data/dataset.py [hunk 2]
```diff
@@ -67,9 +71,8 @@ def _type_masking_task(self, graph: Data):
         self._mask_property(graph, "edge_type", self.__config.task.p_edge, len(EdgeType))
 
     def _token_prediction_task(self, graph: Data):
-        x_mask_value = torch.full((self.__config.max_token_parts,), self.__vocabulary.pad[1], dtype=torch.long)
-        x_mask_value[0] = self.__vocabulary.mask[1]
-        self._mask_property(graph, "x", self.__config.task.p, x_mask_value)
+        mask_value = torch.tensor(self.__tokenizer.encode(MASK).ids, dtype=torch.long)
+        self._mask_property(graph, "x", self.__config.task.p, mask_value)
```
Comment (SokolovYaroslav) [inline src/data/dataset.py:75]: Больше нет `__config.max_token_parts`?

Comment (SokolovYaroslav) [reply src/data/dataset.py:75]: А, оно переехало в `self.__tokenizer.enable_truncation` и `self.__tokenizer.enable_padding`. То есть `self.__tokenizer.encode(MASK).ids` сам западдит?

Comment (SpirinEgor) [reply src/data/dataset.py:75]: Ага, отдельно проверил. У паддинга стоит длина, если последовательность меньше, то он всегда до нее добьет

**Review by SokolovYaroslav — APPROVED** [review]
